test/e2e/specs/editor/blocks/gallery.spec.js

```diff
@@ -74,4 +83,160 @@ test.describe( 'Gallery', () => {
+		await expect( image ).toHaveAttribute( 'src', new RegExp( filename ) );
+
+		const regex = new RegExp(
+			`<!-- wp:gallery {\\"linkTo\\":\\"none\\"} -->\\s*<figure class=\\"wp-block-gallery has-nested-images columns-default is-cropped\\"><!-- wp:image {\\"id\\":\\d+,\\"sizeSlug\\":\\"(?:full|large)\\",\\"linkDestination\\":\\"none\\"} -->\\s*<figure class=\\"wp-block-image (?:size-full|size-large)\\"><img src=\\"[^"]+\/${ filename }\.png\\" alt=\\"\\" class=\\"wp-image-\\d+\\"\/><\/figure>\\s*<!-- \/wp:image --><\/figure>\\s*<!-- \/wp:gallery -->`
+		);
+		await expect.poll( editor.getEditedPostContent ).toMatch( regex );
+	} );
+
+	test( 'gallery caption can be edited', async ( {
+		admin,
+		editor,
+		page,
+	} ) => {
+		const galleryCaption = 'Tested gallery caption';
+
+		await admin.createNewPost();
+		await editor.insertBlock( {
+			name: 'core/gallery',
+			innerBlocks: [
+				{
+					name: 'core/image',
+					attributes: {
+						id: uploadedMedia.id,
+						url: uploadedMedia.source_url,
+					},
+				},
+			],
+		} );
+
+		const gallery = page.locator( 'role=document[name="Block: Gallery"i]' );
+
+		await expect( gallery ).toBeVisible();
+		await editor.selectBlocks( gallery );
+
+		const caption = gallery.locator(
+			'role=textbox[name="Gallery caption text"i]'
+		);
+		await expect( caption ).toBeVisible();
+		await caption.click();
+
+		await page.keyboard.type( galleryCaption );
+
+		await expect
+			.poll( editor.getEditedPostContent )
+			.toMatch(
+				new RegExp( `<figcaption.*?>${ galleryCaption }</figcaption>` )
+			);
+	} );
+
+	test( "uploaded images' captions can be edited", async ( {
+		admin,
+		editor,
+		page,
+	} ) => {
+		const caption = 'Tested caption';
+
+		await admin.createNewPost();
+		await editor.insertBlock( {
+			name: 'core/gallery',
+			innerBlocks: [
+				{
+					name: 'core/image',
+					attributes: {
+						id: uploadedMedia.id,
+						url: uploadedMedia.source_url,
+					},
+				},
+			],
+		} );
+
+		const galleryImage = page.locator(
+			'role=document[name="Block: Gallery"i] >> role=document[name="Block: Image"i]'
+		);
+		const imageCaption = galleryImage.locator(
+			'role=textbox[name="Image caption text"i]'
+		);
+		await expect( galleryImage ).toBeVisible();
+
+		await galleryImage.click();
+		await editor.clickBlockToolbarButton( 'Add caption' );
+
+		await expect( imageCaption ).toBeVisible();
+		await imageCaption.click();
+
+		await page.keyboard.type( caption );
+
+		await expect
+			.poll( editor.getEditedPostContent )
+			.toMatch(
+				new RegExp( `<figcaption.*?>${ caption }</figcaption>` )
+			);
+	} );
+
+	// Disable reason:
+	// This test would be good to enable, but the media modal contains an
+	// invalid role, which is causing Axe tests to fail:
+	// https://core.trac.wordpress.org/ticket/50273
+	//
+	// Attempts to add an Axe exception for the media modal haven't proved
+	// successful:
+	// https://github.com/WordPress/gutenberg/pull/22719
```

Comment:
I wonder of we should copy across the full details of this test so there is a record of what it was supposed to do if the Axe issue can be resolved, eg. the following seems to work locally: 
```
	test.fixme(
		'when initially added the media library shows the Create Gallery view',
		async ( { admin, editor, page } ) => {
			await admin.createNewPost();
			await editor.insertBlock( { name: 'core/gallery' } );
			await page.click( 'role=button[name="Media Library"i]' );
			await page.waitForSelector( '.media-frame' );
			expect( await page.innerText( '.media-frame-title h1' ) ).toBe(
				'Create gallery'
			);
			expect(
				await page.innerText( '.media-toolbar-primary button' )
			).toBe( 'Create a new gallery' );
		}
	);
```

Reply:
Makes sense. Done in [926c24f](https://github.com/WordPress/gutenberg/pull/45202/commits/926c24fad6b7548e08e0fb17b373070154b3ea1d).

Reply:
I don't think we're still running Axe test in Playwright, which is what I missed in the first place. But this probably also means that this test should pass now that Axe is out of the way?

Reply:
I can double-check this and re-enable the test. Thanks for the info, @kevin940726.